FreeRTOS-Plus/Demo/AWS/Fleet_Provisioning_Windows_Simulator/Fleet_Provisioning_With_CSR_Demo/DemoTasks/FleetProvisioningDemoExample.c

```diff
@@ -0,0 +1,864 @@
+ */
+static size_t xThingNameLength;
+
+/**
+ * @brief Buffer to hold responses received from the AWS IoT Fleet Provisioning
+ * APIs. When the MQTT publish callback receives an expected Fleet Provisioning
+ * accepted payload, it copies it into this buffer.
+ */
+static uint8_t pucPayloadBuffer[ democonfigNETWORK_BUFFER_SIZE ];
+
+/**
+ * @brief Length of the payload stored in #pucPayloadBuffer. This is set by the
+ * MQTT publish callback when it copies a received payload into #pucPayloadBuffer.
+ */
+static size_t xPayloadLength;
+
+/**
+ * @brief Global entry time into the application to use as a reference timestamp
+ * in the #prvGetTimeMs function. #prvGetTimeMs will always return the difference
+ * between the current time and the global entry time. This will reduce the chances
+ * of overflow for the 32 bit unsigned integer used for holding the timestamp.
+ */
+static uint32_t ulGlobalEntryTimeMs;
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Each compilation unit that consumes the NetworkContext must define it.
+ * It should contain a single pointer to the type of your desired transport.
+ * When using multiple transports in the same compilation unit, define this pointer as void *.
+ *
+ * @note Transport stacks are defined in FreeRTOS-Plus/Source/Application-Protocols/network_transport.
+ */
+struct NetworkContext
+{
+    TlsTransportParams_t * pxParams;
+};
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Callback to receive the incoming publish messages from the MQTT
+ * broker. Sets xResponseStatus if an expected CreateCertificateFromCsr or
+ * RegisterThing response is received, and copies the response into
+ * responseBuffer if the response is an accepted one.
+ *
+ * @param[in] pPublishInfo Pointer to publish info of the incoming publish.
+ * @param[in] usPacketIdentifier Packet identifier of the incoming publish.
+ */
+static void prvProvisioningPublishCallback( MQTTPublishInfo_t * pPublishInfo,
+                                         uint16_t usPacketIdentifier );
+
+/**
+ * @brief Run the MQTT process loop to get a response.
+ */
+static bool prvWaitForResponse( void );
+
+/**
+ * @brief Subscribe to the CreateCertificateFromCsr accepted and rejected topics.
+ */
+static bool prvSubscribeToCsrResponseTopics( void );
+
+/**
+ * @brief Unsubscribe from the CreateCertificateFromCsr accepted and rejected topics.
+ */
+static bool prvUnsubscribeFromCsrResponseTopics( void );
+
+/**
+ * @brief Subscribe to the RegisterThing accepted and rejected topics.
+ */
+static bool prvSubscribeToRegisterThingResponseTopics( void );
+
+/**
+ * @brief Unsubscribe from the RegisterThing accepted and rejected topics.
+ */
+static bool prvUnsubscribeFromRegisterThingResponseTopics( void );
+
+/**
+ * @brief The timer query function provided to the MQTT context.
+ *
+ * @return Time in milliseconds.
+ */
+static uint32_t prvGetTimeMs( void );
```

Comment:
Is this used?

Reply:
It is no longer used. Removed in this commit: https://github.com/FreeRTOS/FreeRTOS/pull/724/commits/0a18eb6432509f18fd6b1015b07ca343ca5d4e32